packages/amplify-graphql-api-construct-tests/src/__tests__/generations/generation.test.ts

```diff
@@ -0,0 +1,145 @@
+          auth: { apiKey },
+        };
+
+        const variables = {
+          description: `I'd like to make a gluten-free chocolate cake.`,
+        };
+
+        const generateRecipeResult = await doAppSyncGraphqlQuery({ ...args, query: generateRecipe, variables });
+        const recipe = generateRecipeResult.body.data.generateRecipe;
+        expect(recipe.name).toBeDefined();
+      });
+    });
+
+    describe('Generation scalar', () => {
+      test('should generate a string scalar type', async () => {
+        const args = {
+          apiEndpoint,
+          auth: { apiKey },
+        };
+
+        const variables = {
+          input: `
+                Two roads diverged in a yellow wood,
+                And sorry I could not travel both
+                And be one traveler, long I stood
+                And looked down one as far as I could
+                To where it bent in the undergrowth;
+
+                Then took the other, as just as fair,
+                And having perhaps the better claim,
+                Because it was grassy and wanted wear;
+                Though as for that the passing there
+                Had worn them really about the same,
+
+                And both that morning equally lay
+                In leaves no step had trodden black.
+                Oh, I kept the first for another day!
+                Yet knowing how way leads on to way,
+                I doubted if I should ever come back.
+
+                I shall be telling this with a sigh
+                Somewhere ages and ages hence:
+                Two roads diverged in a wood, and I—
+                I took the one less traveled by,
+                And that has made all the difference.`,
+        };
+        const summarizeResult = await doAppSyncGraphqlQuery({ ...args, query: summarize, variables });
+        const summary = summarizeResult.body.data.summarize;
+        expect(summary).toBeDefined();
+      });
+    });
+
+    test('should generate an int scalar type', async () => {
+      const args = {
+        apiEndpoint,
+        auth: { apiKey },
+      };
+
+      const variables = {
+        equation: `
+          There is a three-digit number. The second digit is four times as big as the third digit,
+          while the first digit is three less than the second digit. What is the number?
+        `,
+      };
+      const solveEquationResult = await doAppSyncGraphqlQuery({ ...args, query: solveEquation, variables });
+      const solution = solveEquationResult.body.data.solveEquation;
+      expect(solution).toBeDefined();
+    });
+
+    describe('Generation model', () => {
+      // TODO: This currently doesn't work because LLMs are not great at following regex pattern requirements, they'll sometimes return "<UNKNOWN>"
+      // which fails GraphQL type validation for implicitly generated required model values like id, createdAt, updatedAt.
+      xtest('should generate a model', async () => {
```

Comment:
The test is currently disabled because we throw in the transformer when the return type contains required fields typed as certain AppSync scalars. 

We're exploring options, including prompt improvements, better regex pattern in JSON Schema tool definitions, and special case handling for models (omitting `createdAt`, `updatedAt`, and `id` in tool definition, and populating them in the resolver). 

For now, this is an accepted current limitation.